habitat-lab/habitat/sims/habitat_simulator/debug_visualizer.py

```diff
@@ -344,210 +343,131 @@ def render_debug_frame(
-        additional_savefile_prefix="",
-        debug_lines: Optional[List[Tuple[List[mn.Vector3], mn.Color4]]] = None,
-        debug_circles: Optional[
-            List[Tuple[mn.Vector3, float, mn.Vector3, mn.Color4]]
-        ] = None,
-        show: bool = False,
-    ) -> Tuple[str, List[DebugObservation]]:
-        """
-        Helper function to generate image(s) of the scene for contextual debugging purposes.
-        Specialization to peek a scene. See _peek_bb.
-        Compute a camera placement to view the scene. Show/save an observation. Return the filepath.
+        # first check if the subject is an object id or a string defining ("stage"or"scene") or object handle.
+        if isinstance(subject, int):
+            from habitat.sims.habitat_simulator.sim_utilities import (
+                get_obj_from_id,
+            )
 
-        :param cam_local_pos: Optionally provide a camera location in location local coordinates. Otherwise offset along local -Z axis from the object.
-        :param peek_all_axis: Optionally create a merged 3x2 matrix of images looking at the object from all angles.
-        :param additional_savefile_prefix: Optionally provide an additional prefix for the save filename to differentiate the images.
-        :param debug_lines: Optionally provide a list of debug line render tuples, each with a list of points and a color. These will be displayed in all peek images.
-        :param debug_circles: Optionally provide a list of debug line render circle Tuples, each with (center, radius, normal, color). These will be displayed in all peek images.
-        :param show: If True, open and display the image immediately.
+            subject_obj = get_obj_from_id(self.sim, subject)
+            if subject_obj is None:
+                raise AssertionError(
+                    f"The integer subject, '{subject}', is not a valid object_id."
+                )
+            subject = subject_obj
+        elif isinstance(subject, str):
+            if subject == "stage" or subject == "scene":
+                subject_bb = (
+                    self.sim.get_active_scene_graph()
+                    .get_root_node()
+                    .cumulative_bb
+                )
+                if cam_local_pos is None:
+                    cam_local_pos = mn.Vector3(0, 1, 0)
+            else:
+                from habitat.sims.habitat_simulator.sim_utilities import (
+                    get_obj_from_handle,
+                )
 
-        :return: a tuple containing saved filepath and the list of DebugObservations generated.
-        """
+                subject_obj = get_obj_from_handle(self.sim, subject)
+                if subject_obj is None:
+                    raise AssertionError(
+                        f"The string subject, '{subject}', is not a valid object handle or an allowed alias from ('stage', 'scene')."
+                    )
+                subject = subject_obj
+
+        if subject_bb is None:
```

Comment:
Nit: This could be a `calculate_subject_bounding_box()` method.

Reply:
TODO: I'll refactor this later. I'm adding such a function in another PR.